### PR TITLE
fix: canChangePowerLevel should return true for own user

### PR DESCRIPTION
**Pull request by krille-chan**

### CODE REVIEW

File: lib/src/user.dart [hunk 1]
```diff
@@ -172,7 +172,8 @@ class User extends Event {
   /// Whether the client is allowed to change the power level of this user.
   /// Please be aware that you can only set the power level to at least your own!
   bool get canChangePowerLevel =>
-      room.canChangePowerLevel && powerLevel < room.ownPowerLevel;
+      (room.canChangePowerLevel && powerLevel < room.ownPowerLevel) ||
```
Comment (td-famedly) [inline lib/src/user.dart:175]: I have no idea what this function is supposed to do, how is it different from room.canChangePowerLevel, a rename would be nice.

Also `powerLevel < room.ownPowerLevel` does not make sense

File: lib/src/user.dart [hunk 2]
```diff
@@ -172,7 +172,8 @@ class User extends Event {
...
+      (id == room.client.userID && powerLevel > 0);
```
Comment (td-famedly) [inline lib/src/user.dart:176]: do you actually need `powerLevel > 0`?

File: lib/src/user.dart [hunk 3]
```diff
@@ -172,7 +172,8 @@ class User extends Event {
   /// Whether the client is allowed to change the power level of this user.
   /// Please be aware that you can only set the power level to at least your own!
   bool get canChangePowerLevel =>
```
Comment (td-famedly) [inline lib/src/user.dart:174]: td took way too long to understand what this was supposed to do, can we change this to `canChangeUserPowerLevel` or something better?

File: lib/src/user.dart (hunk 2 above)
Comment (krille-chan) [reply lib/src/user.dart:176]: Hmm in the case this are you self, you can change your own powerlevel only in one direction. So I assumed this makes no sense to return true if the powerlevel is 0

Comment (td-famedly) [reply lib/src/user.dart:176]: yes but you can always create lower power levels, the spec allows you `[-(2**53)+1, (2**53)-1]`

Comment (krille-chan) [reply lib/src/user.dart:176]: okay I checked it out. The correct calculation should be that you:
1. Have the permission to change power levels in general
2. Your power-level is higher than the power level of the target user OR target user is yourself!

File: lib/src/user.dart (hunk 3 above)
Comment (krille-chan) [reply lib/src/user.dart:174]: Okay changed :)

**Pull request merged**